src/common/HTTPResponse.ts

```diff
@@ -78,8 +78,11 @@ export class HTTPResponse {
       ip: responsePayload.remoteIPAddress,
       port: responsePayload.remotePort,
     };
-    // TODO extract statusText from extraInfo.headersText instead if present
-    this._statusText = responsePayload.statusText;
+    if (extraInfo && extraInfo.headersText) {
+      this._statusText = extraInfo.headersText.match(/[^ ]* [^ ]* (.*)\r/)[1];
```

Comment:
does it always match the regex? Or could it be somehow fail if, e.g., a server is misconfigured (configured differently)?

Reply:
I added a test case and improved the parsing, it should not explode if the server sends something weird.

Reply:
it looks it fails on Firefox. Let me land the roll CL first and then we can improve this part.